### ISSUE TITLE
The Screen Recorder says ERROR RECORDER - See Logs for details..

### ISSUE THREAD

**Issue by qnxdev** [EasyScreenCast/EasyScreenCast#257]
Not working, options shows "Somethins gone wrong"

Log is here
--------------------------------------------------------------------------------------------------

The settings of extension EasyScreenCast@iacopodeenosee.gmail.com had an error:
```
ReferenceError: global is not defined
```

Stack trace:
```
TalkativeLog@/home/user1/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/convenience.js:116:9
buildPrefsWidget@/home/user1/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1188:9
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by kedark3** [comment]
I see this same issue as well: Fedora32: 

```The settings of extension EasyScreenCast@iacopodeenosee.gmail.com had an error:

ReferenceError: global is not defined
```

Stack trace:
```
TalkativeLog@/home/user/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/convenience.js:116:9
buildPrefsWidget@/home/user/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1188:9
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by efernandesng** [comment]
See #256

**Comment by kedark3** [comment]
I have tried the solution from #256 but I still keep getting same error. :/

**Comment by matty0blake** [comment]
Not Working it say some thing gone wrong

i think it stop with the new update of gnome i got the same issue


The settings of extension EasyScreenCast@iacopodeenosee.gmail.com had an error:
```
ReferenceError: global is not defined
```

Stack trace:
```
TalkativeLog@/home/julian/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/convenience.js:116:9
buildPrefsWidget@/home/julian/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1188:9
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```



please fix it not recorder work on kali linux expect this one

**Comment by somethingGoneWrong** [comment]
@matty0blake  change line [116 on the conveniece.js](https://github.com/EasyScreenCast/EasyScreenCast/blob/master/convenience.js#L116)
from this `global.log("[ESC]" + msg);`
to this `log("[ESC]" + msg);`
it should work.

**Comment by kedark3** [comment]
@somethingGoneWrong I tried that solution and I see another error: 
```The settings of extension EasyScreenCast@iacopodeenosee.gmail.com had an error:
```
TypeError: listCaps is undefined
```

Stack trace:
```
_updateWebCamCaps@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:969:13
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
_initTabWebcam@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:427:13
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
_init@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:92:18
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
buildPrefsWidget@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1190:18
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```
```

**Comment by somethingGoneWrong** [comment]
@kedark3  try to delete line [969 on pref.js](https://github.com/EasyScreenCast/EasyScreenCast/blob/master/prefs.js#L969) or better move that line inside the below IF. something like this
```
BLABLA
var listCaps = this.CtrlWebcam.getListCapsDevice(device - 1);

if (listCaps !== null && listCaps !== undefined) {
    Lib.TalkativeLog("-^-webcam caps: " + listCaps.length);
    for (var index in listCaps) {
        this.Ref_ListStore_QualityWebCam.set(
            this.Ref_ListStore_QualityWebCam.append(),
            [0],
            [listCaps[index]]
BLABLA
```

**Comment by kedark3** [comment]
Yeah tried that :( didn't work. Seems like the extension is not picking up on changes made as it keeps throwing same error.

**Comment by somethingGoneWrong** [comment]
> it keeps throwing same error.
At the same line? can you show the new log error?

**Comment by kedark3** [comment]
Same Traceback: 
The settings of extension EasyScreenCast@iacopodeenosee.gmail.com had an error:
```
TypeError: listCaps is undefined
```

Stack trace:
```
_updateWebCamCaps@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:969:13
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
_initTabWebcam@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:427:13
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
_init@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:92:18
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
buildPrefsWidget@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1190:18
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by somethingGoneWrong** [comment]
what's on the line **prefs.js:969** now?

**Comment by kedark3** [comment]
I think it finally caught up and now : 
The settings of extension EasyScreenCast@iacopodeenosee.gmail.com had an error:
```
Error: Expected type utf8 for Argument 'str' but got type 'undefined'
```

Stack trace:
```
_updateStateWebcamOptions@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1066:35
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
_init@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:116:18
wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
buildPrefsWidget@/home/kkulkarni/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1175:18
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by watzon** [comment]
I just got this as well, seems to have happened after setting a script that should run after recording is finished 

```
ReferenceError: global is not defined
```

Stack trace:
```
TalkativeLog@/home/watzon/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/convenience.js:116:9
buildPrefsWidget@/home/watzon/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1188:9
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by jmaks** [comment]
the same issue
The Screen Recorder says ERROR RECORDER - See Logs for details.. 

![Screenshot from 2020-06-18 15-46-50](https://user-images.githubusercontent.com/2206770/84999152-f901e980-b17a-11ea-945f-4acc560f4b33.png)

![Screenshot from 2020-06-18 15-45-17](https://user-images.githubusercontent.com/2206770/84999074-e25b9280-b17a-11ea-83bb-418ee3582899.png)


![Screenshot from 2020-06-18 15-43-55](https://user-images.githubusercontent.com/2206770/84998860-9577bc00-b17a-11ea-8567-552728f67828.png)


```
TypeError: listCaps is undefined

Stack trace:
  _updateWebCamCaps@/usr/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:854:13
  wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
  _initTabWebcam@/usr/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:381:13
  wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
  _init@/usr/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:88:18
  wrapper@resource:///org/gnome/gjs/modules/script/_legacy.js:82:27
  buildPrefsWidget@/usr/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/prefs.js:1052:18
  _init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:13
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
  run@resource:///org/gnome/gjs/modules/script/package.js:222:19
  start@resource:///org/gnome/gjs/modules/script/package.js:206:5
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by jmaks** [comment]
after did this mods in files 
**convenience.js** -- change line [116 ](https://github.com/EasyScreenCast/EasyScreenCast/blob/master/convenience.js#L116)
**prefs.js** -- commented string [969](https://github.com/EasyScreenCast/EasyScreenCast/blob/master/prefs.js#L969)

linked comments is --
> @matty0blake change line [116 on the conveniece.js](https://github.com/EasyScreenCast/EasyScreenCast/blob/master/convenience.js#L116)
> from this `global.log("[ESC]" + msg);`
> to this `log("[ESC]" + msg);`
> it should work.

> @kedark3 try to delete line [969 on pref.js](https://github.com/EasyScreenCast/EasyScreenCast/blob/master/prefs.js#L969) or better move that line inside the below IF. something like this
> 

and make one dconf switch screencast recording works, like in [issue 255](https://github.com/EasyScreenCast/EasyScreenCast/issues/255)
>>Workaround:
>dconf write /org/gnome/shell/extensions/EasyScreenCast/device-webcam 0`

it works!

**Comment by jmaks** [comment]
seems, this mods in prefs.js and convenience.js files not usualy needs, or so+^?
@somethingGoneWrong

**Comment by somethingGoneWrong** [comment]
Seems needed on gnome shell 3.36, if you want make a PR and ping @tobias47n9e to update it.

**Comment by msanecki** [comment]
@jmaks thx for the workaround, you save my day!

**Comment by adangel** [comment]
This has been fixed with 1.3.0
